### PR TITLE
Moving all block editor EventListeners into the BlockEditorView.init()

### PR DESCRIPTION
**Pull request by AnmAtAnm**
Also Renamed `BlockEditorController.getDefinitionFormat_()` to `.getBlockFormatCode_()`.

---
This change is [Reviewable](https://reviewable.io/reviews/google/blockly-devtools/290)

### CODE REVIEW

File: src/view/block_editor_view.js [hunk 1]
```diff
@@ -327,11 +343,11 @@ class BlockEditorView {
       }
       this.rtl = newDir;
       this.previewWorkspace = Blockly.inject('preview',
-        {
-          rtl: this.rtl,
-          media: 'media/',
-          scrollbars: true
-        });
+          {
```
Comment (AnmAtAnm) [inline src/view/block_editor_view.js:334]: Indentation correction.

File: src/controller/block_editor_controller.js [hunk 2]
```diff
@@ -355,13 +354,25 @@ class BlockEditorController {
...
   /**
-   * Determines what format the user-defined block definition is in.
+   * Attempt to update the block definition (model, view, and preview), from a
+   * code string.
+   *
+   * @param {string} format The format of the code. Either 'JS' or 'JSON'.
+   * @param {string} code The block definition code.
+   * @return {boolean} True if successful. Otherwise false.
+   */
+  attemptUpdateFromManualCode(format, code) {
```
Comment (AnmAtAnm) [inline src/controller/block_editor_controller.js]: This method will be corrected shortly (now that I understand what it should be doing).

File: src/view/block_editor_view.js [hunk 3]
```diff
@@ -171,24 +171,40 @@ class BlockEditorView {
...
-    let manualBlockDefTA = $('#manualBlockDefTA');
-    manualBlockDefTA.on('input', () => {
-      controller.updatePreview();
+    this.manualBlockDefTA_.on('input', () => {
+      controller.attemptUpdateFromManualCode();
     });
-    manualBlockDefTA.on('keyup', () => {
-      controller.updatePreview();
+    this.manualBlockDefTA_.on('keyup', () => {
+      controller.attemptUpdateFromManualCode();
```
Comment (AnmAtAnm) [inline src/view/block_editor_view.js:190]: attemptUpdateFromManualCode() is just a wrapper for updatePreview() right now.

**Review by picklesrus — COMMENTED** [review]
LGTM

File: src/controller/block_editor_controller.js [hunk 4]
```diff
@@ -155,17 +156,17 @@ class BlockEditorController {
         }
       }
     }
-    // Disable orphans.
-    Blockly.Events.disableOrphans(event);
```
Comment (AnmAtAnm) [inline src/controller/block_editor_controller.js:159]: Move into the view level event listener.

File: src/view/block_editor_view.js [hunk 5]
```diff
@@ -167,28 +167,14 @@ class BlockEditorView {
...
-    });
 
-    const changeFormat = controller.changeFormat.bind(controller);
-    const updatePreview = controller.updatePreview.bind(controller);
-    const refreshPreviews = controller.refreshPreviews.bind(controller);
-
-    // Update code on changes to block being edited.
-    this.editorWorkspace.addChangeListener(() => {
-      controller.updateBlockDefPre();
-      controller.updatePreview();
```
Comment (AnmAtAnm) [inline src/view/block_editor_view.js]: Moved to onWorkspaceChange(..)

File: src/view/block_editor_view.js [hunk 6]
```diff
@@ -167,28 +167,14 @@ class BlockEditorView {
...
+      Blockly.Events.disableOrphans(event);
     });
 
-    // Disable blocks not attached to the factory_base block.
-    this.editorWorkspace.addChangeListener(
-        Blockly.Events.disableOrphans);
-
-    // Update preview on every change.
-    this.editorWorkspace.addChangeListener(
-        refreshPreviews);
```
Comment (AnmAtAnm) [inline src/view/block_editor_view.js]: already part of onWorkspaceChange(..)

File: src/controller/block_editor_controller.js [hunk 7]
```diff
@@ -354,20 +370,11 @@ class BlockEditorController {
   }
 
   /**
-   * Attempt to update the block definition (model, view, and preview), from a
-   * code string.
-   *
-   * @return {boolean} True if successful. Otherwise false.
-   */
-  attemptUpdateFromManualCode() {
```
Comment (AnmAtAnm) [inline src/controller/block_editor_controller.js]: Moved above (a public method)

File: src/controller/block_editor_controller.js [hunk 8]
```diff
@@ -190,13 +191,11 @@ class BlockEditorController {
       const code = blockDefPre.text().trim();
       manualBlockDefTA.val(code);
       manualBlockDefTA.focus();
-      this.updatePreview();
```
Comment (AnmAtAnm) [inline src/controller/block_editor_controller.js:193]: Changing the format should not incur preview update.

**Review by picklesrus — COMMENTED** [review]
lgtm

**Pull request merged**